### ISSUE TITLE
Quickstart?

### ISSUE THREAD

**Issue by AxeOfMen** [sleelin/scimmy-routers#9]
Obviously I'm missing something because after following the sample code, my handlers aren't getting called. Here's my code:

```js
const express = require('express');
const { default: SCIMMYRouters, SCIMMY } = require('scimmy-routers');

const app = express();
const PORT = 3030;

app.use(function(req, res, next) {
  console.log(req.url, req.rawHeaders);
});

SCIMMY.Resources.declare(SCIMMY.Resources.User)
  .ingress((resource, data) => {
    console.log(`Request to create/edit user ${resource}, ${data}`);
  })
  .egress((resource) => {
    console.log(`Request to read user ${resource}`);
  })
  .degress((resource) => {
    console.log(`Request to delete user ${resource}`);
  });

SCIMMY.Resources.declare(SCIMMY.Resources.Group)
  .ingress((resource, data) => {
    console.log(`Request to create/edit group ${resource}, ${data}`);
  })
  .egress((resource) => {
    console.log(`Request to read group ${resource}`);
  })
  .degress((resource) => {
    console.log(`Request to delete group ${resource}`);
  });

let authScheme = {
  type: "bearer",
  handler: (request) => {
    console.log("Inside auth check");
    if (!request.header("Authorization")?.startsWith("Bearer ")) {
      throw new Error("Authorization not detected!");
    } else {
      return 'jkl';
    }
  }
};
app.use("/scim", new SCIMMYRouters(authScheme));

app.listen(PORT, (error) => {
    if (!error)
      console.log("Server is running on port " + PORT)
    else
      console.log("Server cannot start", error);
  }
);

```
I can see incoming requests:
```http
GET /scim/Users?filter=userName+eq+%2278177c22-5b7f-477c-9b25-fe7e7a85ddcc%22 [
  'Authorization',
  'Bearer eyJ...',
  'ADSCIMVersion',
  '6.0',
  'Host',
  'www.mysite.com',
  'Connection',
  'Keep-Alive'
]
```
Nothing else is logged. What am I missing?

**Comment by sleelin** [comment]
Hi @AxeOfMen, thanks for getting in touch!
I was able to reproduce your issue with the code you provided. After a bit more investigation I noticed that the handler for the first call to ``app.use`` doesn't call the ``next()`` method, which means it's holding up express's route handling and not passing the request on to any of the SCIMMY handlers.

When I added a call to the ``next()`` method underneath the first ``console.log``, express proceeded with route handling and the ingress/egress/degress methods were called:
```js
app.use(function(req, res, next) {
  console.log(req.url, req.rawHeaders);
  next()
});
```

With the added call to ``next()``, the console now logs ``Inside auth check``, indicating the auth handler is now running, and when a GET request is made with an Authorization header beginning with "Bearer ", the console additionally logs ``Request to read user [object Object]``, as per the User resource's egress handler.

Let me know if you have any further questions about this, or if the above doesn't resolve the problem for you, so that I can reopen the issue.

Thanks,
Sam
